domain/model/service/service.go

```diff
@@ -542,3 +589,59 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// getWatchActivatedModelsMapper returns a mapper function that filters change events to
+// include only those associated with activated models.
+func getWatchActivatedModelsMapper(st State) func(ctx context.Context, db database.TxnRunner,
+	changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+
+	return func(ctx context.Context, db database.TxnRunner,
+		changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+		modelUUIDs := make([]string, len(changes))
+
+		for i, change := range changes {
+			modelUUID := change.Changed()
+			modelUUIDs[i] = modelUUID
+		}
+
+		// Retrieve all activate status of all models with associated uuids
+		activatedModelUUIDs, err := st.GetActivatedModelUUIDs(ctx, modelUUIDs)
+
+		// There should be no errors returned if there are no activated models found.
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
You should check let of activateModelUUIDS here is equal to zero. If so bail out early.

Reply:
Good point, thanks tom!